### ISSUE TITLE
BulkUpdateWithUndoButton

### ISSUE THREAD

**Issue by friday** [marmelab/react-admin#4474]
**Is your feature request related to a problem? Please describe.**
Currently the only bulk actions are delete actions. I want actions like gmails "Mark as unread", "Mark as Spam", "Archive", "Star". This could also be used as a "soft delete" alternative (setting `deleted` to `true`, then filter out the record).

Related support issue: #3832

**Describe the solution you'd like**
```jsx
<BulkUpdateWithUndoButton {...props} data={{status: 'archived'}} label="Archive" icon={<ArchiveIcon />} />

// Or like this:

<BulkUpdateButton {...props} undoable data={{starred: 'true'}} label="Star" icon={<StarIcon />} />
```
I already implemented it for my project. Most of the work was copied from `BulkDeleteWithUndoButton` replacing Delete with Update, so I think the people who wrote that and know their code may be in a better position to PR, but I'd be happy to share of course.

Update: I've been very busy and haven't made time to understand better what some of the code I copied does (which I prefer to do before I submit a PR) and read/follow your build instructions. If someone is waiting for this, feel free to comment here to remind me.

**Comment by fzaninotto** [comment]
Nice idea, feel free to PR!

**Comment by fzaninotto** [comment]
Fixed by #6072
